src/transformers/tokenization_utils.py

```diff
@@ -1904,6 +1916,22 @@ def prepare_for_model(
 
         # Truncation: Handle max sequence length
         total_len = len_ids + len_pair_ids + (self.num_special_tokens_to_add(pair=pair) if add_special_tokens else 0)
+
+        # Manage padding as multiple of provided integer
+        if pad_to_next_multiple_of is not None:
+            target_length = ((total_len // pad_to_next_multiple_of) + 1) * pad_to_next_multiple_of
```

Comment:
When the `total_len` is already a multiple of the requested value (when 
```
total_len % pad_to_next_multiple_of == 0
```
), I think it will pad to the next multiple also.